Replace cURL with httpClient

TomAbel opened the pull request:
See issue #8 
Reduced code pyramid of doom a little bit via 'errors first' mindset (SmailyForDrupalFormBlock.php)
Tested and ready for review

Commit:
Replace cURL with httpClient

kaittodesk commented on smaily_for_drupal/src/Form/AdminForm.php:
```diff
@@ -154,58 +155,73 @@ public function validateCredentials(array &$form, FormStateInterface $form_state
     }
+    catch (ClientException $e) {
+      switch ($e->getResponse()->getStatusCode()) {
+        case 200:
```
Is there a use case where HTTP 200 status code is returned with `ClientException`?

TomAbel replied:
No, it shouldn't return 200 with an exception. Will patch in next commit

kaittodesk commented on smaily_for_drupal/src/Form/AdminForm.php:
```diff
@@ -154,58 +155,73 @@ public function validateCredentials(array &$form, FormStateInterface $form_state
+        '@password' => $password,
+        '@subdomain' => $subdomain,
+      ];
+      \Drupal::logger('smaily_for_drupal')->error(
```
Logging passwords is a bad practice.

TomAbel replied:
I didn't know what to log so I was more verbose. Should I keep the username/domain or drop it all and have a translateable string?

kaittodesk replied:
Everything else is fine (subdomain and username), but you could also log the HTTP status code. Logging response body could be potentially bad if HTML content is returned.

kaittodesk commented on smaily_for_drupal/src/Form/AdminForm.php:
```diff
@@ -154,58 +155,73 @@ public function validateCredentials(array &$form, FormStateInterface $form_state
+      return $ajax_response;
     }
+    catch (ClientException $e) {
+      switch ($e->getResponse()->getStatusCode()) {
```
You have a lot of repeating code inside this catch block, let's simplify it:

```php
$errorMessage = 'Something went wrong';

// This can also be replaced with if-elseif statement.
switch ($e->getResponse()->getStatusCode()) {
    case 401:
        $errorMessage = 'Credentials invalid.';
        break;
    case 404:
        $errorMessage = 'Subdomain error';
        break;
}

$ajax_response->addCommand(
    new HtmlCommand(
        '.smaily_message',
        '<div class="messages messages--error">' . $errorMessage . '</div>')
);
return $ajax_response;
```

kaittodesk commented on smaily_for_drupal/src/Plugin/Block/SmailyForDrupalFormBlock.php:
```diff
@@ -159,25 +160,27 @@ public function fetchAutoresponders() {
         }
       }
     }
+    catch (ClientException $e) {
```
Try to avoid handling exceptions silently. At least log error to Drupal logger. It will help developers debug errors.

kaittodesk commented on smaily_for_drupal/src/Plugin/Block/SmailyForDrupalFormBlock.php:
```diff
@@ -159,25 +160,27 @@ public function fetchAutoresponders() {
+    $full_url = 'https://' . $domain . '.sendsmaily.net/api/workflows.php?trigger_type=form_submitted';
+    try {
+      $client = \Drupal::httpClient();
+      $request = $client->request('GET', $full_url, [
```
Technically `$client->request(...)` returns response. Use variable `$response` for the return value. If it would return request object that you could use to actually make the request (for example `$client->request(...)->make()`), then it would totally make sense.

Commit:
remove repeating code and password logging

Commit:
Add logging to autoresponder catch, minor changes

Removed random whitespace line
Renamed variable to reduce confusion and to be more explicit

Commit:
Add HTTP status code to log data.

Fixed identation and $subdomain>$domain error.
